modules/standard/Time.chpl

```diff
@@ -1556,20 +1592,28 @@ enum day       { sunday=0, monday, tuesday, wednesday, thursday, friday, saturda
     var adddays = td.days + newhour / 24;
     newhour %= 24;
 
-    return dateTime.combine(date.createFromOrdinal(dt.getDate().toOrdinal()+adddays),
-                            new time(hour=newhour, minute=newmin,
-                                     second=newsec, microsecond=newmicro,
-                                     tz=dt.timezone));
+    if (dt.tz_aware) {
+      return dateTime.combine(date.createFromOrdinal(dt.getDate().toOrdinal()+adddays),
+                              new time(hour=newhour, minute=newmin,
+                                       second=newsec, microsecond=newmicro,
+                                       tz=dt.timezone));
+    } else {
+      return dateTime.combine(date.createFromOrdinal(dt.getDate().toOrdinal()+adddays),
+                              new time(hour=newhour, minute=newmin,
```

Comment:
nitpick, but you could move `date.createFromOrdinal` out of the conditional to reduce the duplication (and make this a little easier on the eyes)